test/e2e/storage/pd.go

```diff
@@ -449,16 +451,38 @@ var _ = SIGDescribe("Pod Disks", func() {
 		// Verify that disk shows up in node 0's volumeInUse list
 		framework.ExpectNoError(waitForPDInVolumesInUse(nodeClient, diskName, host0Name, nodeStatusTimeout, true /* should exist*/))
 
-		output, err := exec.Command("gcloud", "compute", "instances", "list", "--project="+framework.TestContext.CloudConfig.ProjectID).CombinedOutput()
+		// Initialize GCE API
+		ctx := context.Background()
```

Comment:
The test infra already provides a gceCloud object that should be doing all this setup.  Could you look into using those library methods?

Reply:
I was actually unable to find a library function that does this setup. I found getComputeClient() in /test/e2e_node/runner/remote/run_remote.go but it doesn't do exactly what I need and doesn't provide a context to work with.

I also ran into a test, testStackdriverMonitoring, with similar requirements in /test/e2e/instrumentation/monitoring/stackdriver.go that just created their own client like I did.

Could you point me to the library method you're referring to? Or please advise on next steps.

Looks like if we can't find a  library method maybe we should create one? The existing one that is "similar" specifically has a retry loop because apparently getting these credentials on jenkins is flaky. We should probably also fix/notify the stackdriver one about fixing their potentially flaky test.

Reply:
The e2e test should already do its own cloud provider initialization during the init.  If you call `GetGCECloud()`, it will return you an object that should have the credentials already set up.  It may be that the library doesn't have a method for the command we want, but we can add whatever we need.